elytron/WFCORE-4150-credential-store-automatic-updates.adoc

```diff
@@ -0,0 +1,125 @@
+perform.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4150[WFCORE-4150]
+* https://issues.jboss.org/browse/WFLY-12218[WFLY-12218]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-676[EAP7-676]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:jkasik@redhat.com[Jan Kasik]
+
+=== Testing By
+
+[ ] Engineering
+[x] QE
+
+=== Affected Projects or Components
+
+* WildFly Core
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+It should be possible for a ```credential-reference``` to support both a credential store (i.e., the ```store```
+attribute) and a clear text password (i.e., the ```clear-text``` attribute) to be specified at the same time.
+
+When adding a new ```credential-reference``` with both the ```store``` and ```clear-text``` attributes specified:
+
+* If the ```alias``` attribute is also specified, one of the following will then occur:
+
+** If the credential store does not contain an entry for the given alias, a new entry will be added to the credential
+store to hold the clear text password that was specified. The ```clear-text``` attribute will then be removed from the
+management model.
+
+** If the credential store does contain an entry for the given alias, the existing credential will be replaced with
+the clear text password that was specified. The ```clear-text``` attribute will then be removed from the management
+model.
+
+* If the ```alias``` attribute is not specified, an alias will be generated and a new entry will be added to the
+credential store to hold the clear text password that was specified. The ```clear-text``` attribute will then be
+removed from the management model.
+
+Note that if only the ```clear-text``` attribute is specified or if only the ```store``` and ```alias``` attributes
+are specified when adding a new ```credential-reference```, there will be no changes in behaviour.
+
+When updating an existing ```credential-reference``` that contains both the ```alias``` and ```store``` attributes to
+also specify the ```clear-text``` attribute:
+
+* The existing credential will be replaced with the clear text password that was specified. The ```clear-text```
+attribute will then be removed from the management model.
+
+In each of the above cases where adding or updating a ```credential-reference``` results in a new entry being added
+to the credential store or an existing entry being updated in the credential store, the output of the management
+operation will indicate this.
+
+==== Rollback upon operation failure
+
+If an operation that includes a ```credential-reference``` parameter fails for any reason, no automatic
+credential store update will take place, i.e., any credential store that was specified via the
+```credential-reference``` attribute will contain the same contents as it did before the operation was
```

Comment:
Right now the document renders in a weird way - this line is missing and rest of the document is all in a grey code block.
It can be fixed by putting credential-reference into just one apostrophe instead of three.

Reply:
Fixed, thanks!